services-custom/s3-transfermanager/src/main/java/software/amazon/awssdk/custom/s3/transfer/internal/S3CrtUtils.java

```diff
@@ -238,19 +237,28 @@ private static String encodedQueryString(Map<String, List<String>> rawQueryParam
                            .orElse("");
     }
 
-    private static void addRequestOverrideConfiguration(AwsRequestOverrideConfiguration requestOverrideConfiguration,
-                                                        Consumer<HttpHeader[]> headersConsumer,
-                                                        Consumer<String> queryParametersConsumer) {
-        throwExceptionForUnsupportedConfigurations(requestOverrideConfiguration);
+    private static void processRequestOverrideConfiguration(Optional<AwsRequestOverrideConfiguration> optionalConfig,
+                                                            Consumer<String> queryParametersConsumer) {
+        if (optionalConfig.isPresent()) {
+            AwsRequestOverrideConfiguration requestOverrideConfiguration = optionalConfig.get();
 
-        if (!requestOverrideConfiguration.headers().isEmpty()) {
-            HttpHeader[] httpHeaders = createHttpHeaders(requestOverrideConfiguration.headers());
-            headersConsumer.accept(httpHeaders);
+            throwExceptionForUnsupportedConfigurations(requestOverrideConfiguration);
+            if (!requestOverrideConfiguration.rawQueryParameters().isEmpty()) {
+                String encodedQueryString = encodedQueryString(requestOverrideConfiguration.rawQueryParameters());
+                queryParametersConsumer.accept(encodedQueryString);
+            }
         }
+    }
 
-        if (!requestOverrideConfiguration.rawQueryParameters().isEmpty()) {
-            String encodedQueryString = encodedQueryString(requestOverrideConfiguration.rawQueryParameters());
-            queryParametersConsumer.accept(encodedQueryString);
+    private static void addCustomHeaders(Optional<AwsRequestOverrideConfiguration> requestOverrideConfiguration,
```

Comment:
Using `Optional` as input param might be a bit controversial; I can't think of better ways for this method and it's a private method.

Reply:
Yeah, I thought about doing null check in this method because it feels a bit unnecessary with the existing optional object. But I agree using optional here is a code smell. I'll update it.